### ISSUE TITLE
New candidate revision (r57) create icons issue

### ISSUE THREAD

**Issue by seb128** [ubuntu/gnome-sdk#257]
Testing r57 with the evince candidate the icons in the decoration are missing, see screenshot

![evince](https://github.com/user-attachments/assets/9da5c95a-57c5-4c0f-b7e1-57afa5c11e65)

Also trying gnome-calculator from candidate

```
$ /snap/bin/gnome-calculator 
...
(gnome-calculator:22523): Gtk-WARNING **: 09:10:27.684: Failed to load icon /snap/gnome-calculator/965/data-dir/icons/Yaru/scalable/actions/edit-undo-symbolic.svg: Unable to load image-loading module: /snap/gnome-calculator/965/gnome-platform/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /snap/gnome-calculator/965/gnome-platform/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: cannot open shared object file: No such file or directory
```

**Comment by seb128** [comment]
In r48
/snap/gnome-46-2404/48/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders.cache has 
"/build/gnome-46-2404-sdk/prime/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so"

in r57
/snap/gnome-46-2404/57/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders.cache
"/build/gnome-46-2404-sdk/prime/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader_svg.so"


libpixbufloader-svg.so change to libpixbufloader_svg.so for some reason?

**Comment by sergio-costas** [comment]
I'll check it.

**Comment by sergio-costas** [comment]
Mmm... odd... I'll try to rename that file.

**Comment by sergio-costas** [comment]
Ok, it seems that snaps that use gnome extension do rebuild the gdk-pixbuf-loaders.cache at install or update. The problem is that, when the gnome-XX-XXXX.snap is updated, that cache isn't rebuilt, so any change would break that.

Renaming the loader in the sdk and in the contents snap does work, but the best solution would be to rebuild all those caches (not only the loaders, but also the icons and others) not only when a snap is updated, but also when the extensions on which it depends are updated.

**Comment by seb128** [comment]
Is that rebuild logic part of the gnome wrapper?

**Comment by sergio-costas** [comment]
Maybe the best course of action is to check if `libpixbufloader-svg.so` does exist, and if not, do a soft link from `libpixbufloader_svg.so`. What do you think?

**Comment by sergio-costas** [comment]
Yes, it is. IIRC, it is part of the snapcraft-desktop-integration scripts.

**Comment by sergio-costas** [comment]
Here: https://github.com/canonical/snapcraft-desktop-integration/blob/80aefcc8acafa2656e666ffa5e905fd91e2999e3/common/desktop-exports#L435

**Comment by sergio-costas** [comment]
The `needs_update` variable is set in these and more lines: https://github.com/canonical/snapcraft-desktop-integration/blob/80aefcc8acafa2656e666ffa5e905fd91e2999e3/common/init#L32

**Comment by seb128** [comment]
So I think we need to update the wrapper logic first. I would suggest we revert the gnome-sdk change at least until the wrapper change is landed and published to stable

**Comment by sergio-costas** [comment]
Here it uses a md5sum to detect changes in the gtk-common-themes snap. Maybe we could do something like that for gnome? https://github.com/canonical/snapcraft-desktop-integration/blob/80aefcc8acafa2656e666ffa5e905fd91e2999e3/common/init#L46

**Comment by sergio-costas** [comment]
The point is that I didn't change anything that whould produce that. It seems something in gdk-pixbuf-loaders, or maybe librsvg...

**Comment by seb128** [comment]
The issue was resolved since
